aten/src/ATen/core/function_schema.cpp

```diff
@@ -17,6 +17,22 @@ const std::vector<Argument>& FunctionSchema::getCorrectList(SchemaArgType type)
   }
 }
 
+FunctionSchema FunctionSchema::cloneWithRealTypes() const {
+  auto cloneWithRealTypes = [](const Argument& a) {
+    return a.cloneWithType(a.real_type() ? a.real_type() : a.type());
```

Comment:
It might be a good idea to enforce `real_type()` is always non-null and points to `type()`, just feeling lazy...